concordium-sdk/src/main/java/com/concordium/sdk/Client.java

```diff
@@ -303,6 +304,28 @@ public PeerStatistics getPeerStatistics(final boolean includeBootstrappers) {
         return PeerStatistics.parse(value);
     }
 
+    /**
+     * Get the IDs of the bakers registered in the given block.
+     * Note that this list is in general different from the bakers that are
+     * returned as part of [get_birk_parameters](Client::get_birk_parameters).
+     * The latter are the bakers that are eligible for baking at the time of
+     * the block.
+     * @param blockHash {@link Hash} of the block bakers are to be retrieved.
+     * @return Parsed {@link ImmutableList} of {@link BakerId}
+     * @throws Exception When the returned JSON is null.
+     */
+    public ImmutableList<BakerId> getBakerList(Hash blockHash) throws Exception {
+        val res = server()
+                .getBakerList(ConcordiumP2PRpc.BlockHash.newBuilder().setBlockHash(blockHash.asHex()).build());
+
+        BakerId[] bakerIdArray = BakerId.fromJsonArray(res);
+        if (Objects.isNull(bakerIdArray)) {
+            throw new Exception(String.format("Could not retrieve Baker Ids at block %s", blockHash.asHex()));
```

Comment:
```suggestion
            throw new Exception(String.format("Could not retrieve baker ids for block %s", blockHash.asHex()));
```
Also please use `BlockNotFoundException ` as mentioned above.

Reply:
`BlockNotFoundException` currently does not have a string constructor argument. Should we add one?

Reply:
It takes a `Hash` which I believe exactly is what `blockHash` is? So if you just omit the `.asHex()` I think you should be able to use it.

Reply:
Done, This has been refactored to just use block hash. Please do check